File: src/2348. Number of Zero-Filled Subarrays.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function zeroFilledSubarray(nums) {
  let left = 0
  let count = 0
  for (let right = 0; right <= nums.length; ++right) {
    if (right === nums.length || nums[right] !== 0) {
      const n = right - left
      count += (n * (n + 1)) / 2
      left = right + 1
    }
  }
  return count
}

File: src/2348. Number of Zero-Filled Subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int left = 0;
        long long count = 0;
        for (int right = 0; right <= nums.size(); ++right) {
            if (right == nums.size() || nums[right] != 0) {
                int n = right - left;
                count += (long long) n * (n + 1) / 2;
                left = right + 1;
            }
        }
        return count;
    }
};